test/system/deposit-liquidated-externally.test.js

```diff
@@ -0,0 +1,153 @@
+const { expect } = require("chai")
+const {
+  to1e18,
+  impersonateAccount,
+  resetFork,
+  ZERO_ADDRESS,
+} = require("../helpers/contract-test-helpers")
+const Auction = require("../../artifacts/contracts/Auction.sol/Auction.json")
+
+const describeFn =
+  process.env.NODE_ENV === "system-test" ? describe : describe.skip
+
+// This system test scenario checks the behaviour of Coverage Pools when the
+// deposit on auction has been liquidated outside Coverage Pools. It is meant
+// to be executed on Hardhat Network with mainnet forking enabled.
+// At the start, the fork is being reset to the specific starting block which
+// determines the initial test state. This test uses real mainnet contracts.
+// There are two test cases: one for the auction partially filled and one for
+// the auction fully filled. In both cases there is an attempt to take offer on
+// an auction with an invalid state deposit, because the deposit has been
+// liquidated between auction creating and taking offer. In both cases the
+// transaction should be reverted
+describeFn("System -- deposit liquidated outside Coverage Pools", () => {
+  const startingBlock = 12368838
+  const tbtcTokenAddress = "0x8daebade922df735c38c80c7ebd708af50815faa"
+  const depositAddress = "0x55d8b1dd88e60d12c81b5479186c15d07555db9d"
+  const bidderAddress = "0xa0216ED2202459068a750bDf74063f677613DA34"
+  const keepTokenAddress = "0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC"
+  const tbtcDepositTokenAddress = "0x10b66bd1e3b5a936b7f8dbc5976004311037cdf0"
+  const auctionLength = 86400 // 24h
+  // Only deposits with at least 50% of bonds offered on bond auction will be
+  // accepted by the risk manager.
+  const bondAuctionThreshold = 50
```

Comment:
Not a big deal but maybe we can set it to 66% and comment that TBTC offers 66.6667% of singer bond when opening an auction? It is easier to explain and is closer to the real-world scenario.

Reply:
Fixed in a separate PR https://github.com/keep-network/coverage-pools/pull/110/